src/pocketmine/form/MenuForm.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\form;
+
+use pocketmine\Player;
+
+/**
+ * This form type presents a menu to the user with a list of options on it. The user may select an option or close the
+ * form by clicking the X in the top left corner.
+ */
+abstract class MenuForm extends Form{
+
+	/** @var string */
+	protected $content;
+	/** @var MenuOption[] */
+	private $options;
+
+	/** @var int|null */
+	private $selectedOption;
+
+	/**
+	 * @param string       $title
+	 * @param string       $text
+	 * @param MenuOption[] $options
+	 */
+	public function __construct(string $title, string $text, MenuOption ...$options){
+		parent::__construct($title);
+		$this->content = $text;
+		$this->options = $options;
+	}
+
+	public function getType() : string{
+		return Form::TYPE_MENU;
+	}
+
+	public function getOption(int $position) : ?MenuOption{
+		return $this->options[$position] ?? null;
+	}
+
+	/**
+	 * Returns the index of the option selected by the user. Pass this to {@link getButton} to get the button object
```

Comment:
Dead link to `getButton`. I believe this should be `Pass this to {@link MenuForm::getOption} to get the MenuOption object`.